pkg/notifications/slack/slack.go

```diff
@@ -43,10 +43,10 @@ func notify(hookURL, username, channel, message string) error {
 	return nil
 }
 
-func NotifyDrain(hookURL, username, channel, nodeID string) error {
-	return notify(hookURL, username, channel, fmt.Sprintf("Draining node %s", nodeID))
+func NotifyDrain(hookURL, username, channel, messageTemplate, nodeID string) error {
```

Comment:
These 2 functions are now identical. I Didn't want to change the API as part of this PR, but I am also happy to refactor. For example:

- I could move `fmt.Sprintf(messageTemplate, nodeID)` to the place in main.go where the notification is sent, leaving in this file a single function `Notify`.
- I could also add a struct to this package, which will contain all the slack options (URL, channel, user, message templates). The struct would be created once from main (similar to `DaemonSetLock`). The methods `NotifyDrain` and `NotifyReboot` will be added tot he struct and only need to receive the `nodeID`.

The second refactor is a bit more involved but feels like a step in the direction of a more flexible/generic notifications approach.

Reply:
You will need someone with more experience than me in this project to review whether this API was used outside the current code.

IMO, we should probably get rid of it.

Nevertheless, this will probably be modified if we go to a generic notification system, which will probably remove a bunch of code, including this one. So I wouldn't tackle this abstraction in here, and would keep the current code. At best, I would indeed expose notify as `Notify` and use it in main, but I don't see it as a necessity myself.

I would welcome you to join the effort of the generic notification system refactoring though, your review could be extremely valuable! :)

Reply:
yeah makes total sense to leave further changes outside of this PR and tackle as a proper refactoring!